engine/render/src/render/font_renderer.cpp

```diff
@@ -1047,15 +1048,23 @@ namespace dmRender
             case dmRender::RENDER_LIST_OPERATION_END:
                 {
                     uint32_t buffer_size = sizeof(GlyphVertex) * text_context.m_VertexIndex;
-                    dmGraphics::SetVertexBufferData(text_context.m_VertexBuffer, 0, 0, dmGraphics::BUFFER_USAGE_STREAM_DRAW);
-                    dmGraphics::SetVertexBufferData(text_context.m_VertexBuffer, buffer_size, text_context.m_ClientBuffer, dmGraphics::BUFFER_USAGE_STREAM_DRAW);
-                    text_context.m_VerticesFlushed = text_context.m_VertexIndex;
+                    dmhash_t hash = dmHashBuffer64(text_context.m_ClientBuffer, buffer_size);
+                    if (hash != text_context.m_PreviousHash)
```

Comment:
Create a hash of the vertex data we want to upload.
If it differs from the previously uploaded data, then we upload again.

Reply:
The way it differs from the previous attempt is that this always checks the entire buffer whilst the last "fix" only checked if the number of vertices were the same (! which of cours was very silly of me).